pkg/nfd/nodefeaturerule_test.go

```diff
@@ -0,0 +1,229 @@
+package nfd_test
```

Comment:
```suggestion
package nfd
```
We generally keep tests in the same package so that we can check errorMsg directly, for example